crates/pathfinder/src/consensus/inner/consensus_task.rs

```diff
@@ -54,22 +56,29 @@ pub fn spawn(
     util::task::spawn(async move {
         // Get the validator address and validator set provider
         let validator_address = config.my_validator_address;
-        let validator_set_provider = L2ValidatorSetProvider::new(storage.clone(), chain_id, config);
-
-        let mut consensus = Consensus::recover(
-            Config::new(validator_address)
-                .with_wal_dir(wal_directory)
-                .with_history_depth(
-                    // TODO: We don't support round certificates yet, and we want to limit
-                    // rebroadcasting to a minimum. Rebroadcast timeouts will happen for historical
-                    // engines which are finalized because the effect `CancelAllTimeouts` is only
-                    // triggered upon a new round or a new height.
-                    0,
-                ),
-            // TODO use a dynamic validator set provider, once fetching the validator set from the
-            // staking contract is implemented. Related issue: https://github.com/eqlabs/pathfinder/issues/2936
-            Arc::new(validator_set_provider.clone()),
-        )?;
+        let validator_set_provider =
+            L2ValidatorSetProvider::new(storage.clone(), chain_id, config.clone());
+
+        // Enforce a fixed proposer for now
+        let proposer_selector = FixedProposerSelector::new(config.proposer_address);
```

Comment:
I think I'd prefer the round-robin by default (especially for the smoke test), IOW making the `proposer_address` optional...

Reply:
Yes, that's what I meant.